packages/xrpl/test/integration/utils.ts

```diff
@@ -1,45 +1,198 @@
+async function sendLedgerAccept(client: Client): Promise<unknown> {
+  return client.connection.request({ command: 'ledger_accept' })
+}
+
+/**
+ * Throttles an async function in a way that can be awaited.
+ * By default throttle doesn't return a promise for async functions unless it's invoking them immediately.
+ *
+ * @param func - async function to throttle calls for.
+ * @param wait - same function as lodash.throttle's wait parameter. Call this function at most this often.
+ * @returns a promise which will be resolved/ rejected only if the function is executed, with the result of the underlying call.
+ */
+// eslint-disable-next-line @typescript-eslint/no-explicit-any -- Proper
+function asyncThrottle<F extends (...args: any[]) => Promise<unknown>>(
+  func: F,
+  wait?: number,
+): (...args: Parameters<F>) => ReturnType<F> {
+  const throttled = throttle((resolve, reject, args: Parameters<F>) => {
+    func(...args)
+      .then(resolve)
+      .catch(reject)
+  }, wait)
+  const ret = (...args: Parameters<F>): ReturnType<F> =>
+    new Promise((resolve, reject) => {
+      throttled(resolve, reject, args)
+    }) as ReturnType<F>
+  return ret
 }
 
-export function subscribeDone(client: Client, done: Mocha.Done): void {
+const throttledLedgerAccept = asyncThrottle(sendLedgerAccept, 1000)
+
+export async function ledgerAccept(
+  client: Client,
+  retries?: number,
+  shouldThrottle?: boolean,
+): Promise<unknown> {
+  return new Promise<unknown>((resolve, reject) => {
+    const ledgerAcceptFunc = shouldThrottle
+      ? throttledLedgerAccept
+      : sendLedgerAccept
+    ledgerAcceptFunc(client)
+      .then(resolve)
+      .catch((error) => {
+        if (retries === undefined) {
+          setTimeout(() => {
+            resolve(ledgerAccept(client, 10))
+          }, 1000)
+        } else if (retries > 0) {
+          setTimeout(() => {
+            resolve(ledgerAccept(client, retries - 1))
+          }, 1000)
+        } else {
+          reject(error)
+        }
+      })
+  })
+}
+
+/**
+ * Attempt to get the time after which we can check for the escrow to be finished.
+ * Sometimes the ledger close_time is in the future, so we need to wait for it to catch up.
+ *
+ * @param targetTime - The target wait time, before accounting for current ledger time.
+ * @param minimumWaitTimeInMs - The minimum wait time in milliseconds.
+ * @param maximumWaitTimeInMs - The maximum wait time in milliseconds.
+ * @returns The wait time in milliseconds.
+ */
+export function calculateWaitTimeForTransaction(
+  targetTime: number,
+  minimumWaitTimeInMs = 5000,
+  maximumWaitTimeInMs = 20000,
+): number {
+  const currentTimeUnixInMs = Math.floor(new Date().getTime())
+  const currentTimeRippleInSeconds = unixTimeToRippleTime(currentTimeUnixInMs)
+  const closeTimeCurrentTimeDiffInSeconds =
+    currentTimeRippleInSeconds - targetTime
+  const closeTimeCurrentTimeDiffInMs = closeTimeCurrentTimeDiffInSeconds * 1000
```

Comment:
Can the `In` be removed from these constants? It makes them harder to read IMO

Reply:
Done